openaev-api/src/main/java/io/openaev/service/connectors/XtmComposerService.java

```diff
@@ -0,0 +1,208 @@
+    Map<String, String> composerSettings = new HashMap<>();
+    composerSettings.put(XTM_COMPOSER_ID.key(), input.getId());
+    composerSettings.put(XTM_COMPOSER_NAME.key(), input.getName());
+    composerSettings.put(
+        XTM_COMPOSER_VERSION.key(), this.platformSettingsService.getPlatformVersion());
+    composerSettings.put(XTM_COMPOSER_PUBLIC_KEY.key(), input.getPublicKey());
+    composerSettings.put(
+        XTM_COMPOSER_LAST_CONNECTIVITY_CHECK.key(), LocalDateTime.now().toString());
+
+    Map<String, Setting> savedSettings = platformSettingsService.saveSettings(composerSettings);
+
+    return toXtmComposerOutput(
+        savedSettings.get(XTM_COMPOSER_ID.key()).getValue(),
+        savedSettings.get(XTM_COMPOSER_VERSION.key()).getValue());
+  }
+
+  /**
+   * Refresh connectivity with XTM Composer by updating last connectivity check time
+   *
+   * @param xtmComposerId XTM Composer id
+   * @param lastConnectivityCheck Last connectivity check time
+   * @return XtmComposerOutput
+   */
+  public XtmComposerOutput refreshConnectivity(
+      String xtmComposerId, LocalDateTime lastConnectivityCheck) {
+    Map<String, Setting> xtmComposerInformation = this.getXtmComposerSettings();
+    if (!xtmComposerId.equals(xtmComposerInformation.get(XTM_COMPOSER_ID.key()).getValue())) {
+      throw new BadRequestException("Invalid xtm-composer identifier");
+    }
+
+    platformSettingsService.saveSetting(
+        XTM_COMPOSER_LAST_CONNECTIVITY_CHECK.key(), lastConnectivityCheck.toString());
+
+    return toXtmComposerOutput(
+        xtmComposerInformation.get(XTM_COMPOSER_ID.key()).getValue(),
+        xtmComposerInformation.get(XTM_COMPOSER_VERSION.key()).getValue());
+  }
+
+  /**
+   * Check if the last connectivity check is older than 1 day
+   *
+   * @param lastConnectivityCheckValue Last connectivity check value as string
+   * @return true if the last connectivity check is older than 1 day, false otherwise
+   */
+  public boolean isLastConnectivityCheckTooOld(String lastConnectivityCheckValue) {
+    try {
+      LocalDateTime lastCheck = LocalDateTime.parse(lastConnectivityCheckValue);
+      LocalDateTime oneDayAgo = LocalDateTime.now().minusDays(1);
+      return lastCheck.isBefore(oneDayAgo);
+    } catch (Exception e) {
+      log.error("Error parsing last connectivity check value: {}", e.getMessage());
+      return false;
+    }
+  }
+
+  /**
+   * Validate XTM Composer identifier
+   *
+   * @param xtmComposerId XTM Composer id to validate
+   */
+  public void validateXtmComposerId(String xtmComposerId) {
+    Map<String, Setting> xtmComposerInformation = this.getXtmComposerSettings();
+    if (!xtmComposerId.equals(xtmComposerInformation.get(XTM_COMPOSER_ID.key()).getValue())) {
+      throw new BadRequestException("Invalid xtm-composer identifier");
+    }
+  }
+
+  /** Validate XTM Composer reachability */
+  public void validateXtmComposerReachability() {
```

Comment:
**nitpick :** Maybe it would make sense to return a boolean instead of returning nothing and relying on the caller expecting an exception that is not in the signature of the method ?

Reply:
> Actually the endpoint already return a boolean


That's what confuses me. Why don't you return a boolean in validateXtmComposerReachability instead of catching a generic exception in the caller function ? You could then just return the result of validateXtmComposerReachability in isXtmComposerReachable. 
Even if that bother you to return a boolean, we should at least have a signature that explicitly says that we throw an exception so that it's explicit what exception can happen.
```suggestion
  public void validateXtmComposerReachability() throws BadRequestException {
```